### PR TITLE
[TEP074] Remove Pullrequest pipelineResources

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes
This commit removes the PullRequest Resources support.
This PR removes `github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1/pullrequest`
and docs for PullRequest resources.

Removal of `pipelineResources`, as in tektoncd#5967 has been broken up into removal
of each resources packages for the `small PR` code standard.

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: please migrate off of `pullrquests` `pipelineresources` as it is removed, please refer to the doc at https://github.com/tektoncd/pipeline/blob/main/docs/pipelineresources.md#replacing-an-pullrequest-resource
```

### CODE REVIEW

**Comment by afrittoli** [comment]
Note that [this task](https://hub.tekton.dev/tekton/task/pull-request) depends on the container images used by the PullRequestPipelineResource. The container image will of course continue to exist, but it won't be possible to update it in future with security patches, so we need to decide whether to deprecate the task as well, rewrite it to use a different image or move the existing code somewhere else.

I would vote to deprecate the task.

**Comment by dibyom** [comment]
If we get a maintainer for the pull_request task maybe they can maintain the image? (or switch to a different image?)

**Comment by QuanZhang-William** [comment]
> Note that [this task](https://hub.tekton.dev/tekton/task/pull-request) depends on the container images used by the PullRequestPipelineResource. The container image will of course continue to exist, but it won't be possible to update it in future with security patches, so we need to decide whether to deprecate the task as well, rewrite it to use a different image or move the existing code somewhere else.
> 
> I would vote to deprecate the task.

As we are migrating the catalogs to different repos and implementing support tiers, technically only the `verified catalogs` (golang-build, golang-test, git-clone, git-batch-merge and kaniko) will be  owned and maintained by @tektoncd/catalog-maintainers. And we will send notice to the owners of other resources in https://github.com/tektoncd/catalog to migrate their resources to their own repos.

The [git-init](https://github.com/tektoncd/pipeline/tree/main/cmd/git-init) images will be used by `git-batch-merge` and `git-clone` tasks, so I will extract the [git-init](https://github.com/tektoncd/pipeline/tree/main/cmd/git-init) to the new [git-clone](https://github.com/tektoncd-catalog/git-clone) catalog repo as part of the migration.

**Comment by lbernick** [comment]
It looks like the task hasn't gotten updates in about a year and doesn't have a readme, so I'd vote to deprecate the task as well. FYI task owners @bobcatfish and @wlynch-- I'm guessing you're not actively maintaining this task?

@JeromeJu can you please update https://github.com/tektoncd/pipeline/blob/main/tekton/publish.yaml#L14 and https://github.com/tektoncd/pipeline/blob/main/docs/install.md#verify-the-transparency-logs-using-rekor-cli as well?

**Comment by lbernick** [comment]
@afrittoli @dibyom I just realized the TEP proposes moving these images into a new repo tektoncd/images (https://github.com/tektoncd/community/blob/main/teps/0074-deprecate-pipelineresources.md#proposal) but I'm questioning the value of this; should we just update the TEP to specify that the images can be deleted?

File: docs/pipeline-api.md [hunk 1]
```diff
@@ -4569,6 +4569,9 @@ reasons that emerge from underlying resources are not included here</p>
 </tr><tr><td><p>&#34;TaskRunImagePullFailed&#34;</p></td>
 <td><p>TaskRunReasonImagePullFailed is the reason set when the step of a task fails due to image not being pulled</p>
 </td>
+</tr><tr><td><p>&#34;TaskRunResultLargerThanAllowedLimit&#34;</p></td>
```
Comment (lbernick) [inline docs/pipeline-api.md:4572]: Can you split this into a separate PR please?

File: pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go [hunk 2]
```diff
@@ -136,32 +122,6 @@ func TestResourceValidation_Invalid(t *testing.T) {
 	}
 }
 
-func TestResourceValidation_Valid(t *testing.T) {
```
Comment (XinruZhang) [inline pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go:139]: If we remove this test, there's no test cases to ensure the behavior of Valid ResourceValidation... Is the plan to remove all kinds of pipelineresources (guess so?), what's the estimated timeline?

Without adding unit tests to the PipelineResource validation, we may end up in an unreleasable state :(

Comment (JeromeJu) [reply pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go:139]: This test case has been gradually removed along with the removal of each pipelineResources eg. [cloudEvent resources](https://github.com/tektoncd/pipeline/pull/5995) and [clusterResources](https://github.com/tektoncd/pipeline/pull/5996). And we are targeting to remove all pipelineResources within the next release so I think this tests now serves not the exact purpose as it did previously.

File: docs/pipeline-api.md (hunk 1 above)
Comment (JeromeJu) [reply docs/pipeline-api.md]: Thanks. this was fixed by rebasing.

File: pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go (hunk 2 above)
Comment (JeromeJu) [reply pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go:139]: Plus I think we might have already not been updating v1alpha1 as they were not supported, it might be safe to assume that we can delete this?

Comment (XinruZhang) [reply pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go:139]: SG, I'm okay If we are targeting at deleting everything within this release, we should update https://github.com/tektoncd/pipeline/issues/5967 to reflect that and add that issue to release v0.45.0

Comment (JeromeJu) [reply pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go:139]: Thanks @XinruZhang , updated at https://github.com/tektoncd/pipeline/issues/5967#issuecomment-1405011306

**Review by lbernick — APPROVED** [review]
can you double check that there are no pullrequest pipelineresource docs that need to be removed?

**Comment by JeromeJu** [comment]
> can you double check that there are no pullrequest pipelineresource docs that need to be removed?

Thanks, updated!

**Comment by Yongxuanzhang** [comment]
I'm curious if https://github.com/tektoncd/pipeline/tree/main/cmd/pullrequest-init and https://github.com/tektoncd/pipeline/tree/main/pkg/pullrequest are related to this tep?

**Comment by JeromeJu** [comment]
> I'm curious if https://github.com/tektoncd/pipeline/tree/main/cmd/pullrequest-init and https://github.com/tektoncd/pipeline/tree/main/pkg/pullrequest are related to this tep?
> 
> Oh I see you removed it in another PR

Yes they are the images that uses the pullrequest images and tried to separate them for smaller PRs :)
